packages/material-ui/src/internal/SwitchBase.js

```diff
@@ -33,6 +33,23 @@ const SwitchBaseRoot = styled(ButtonBase, { skipSx: true })(({ styleProps }) =>
   ...(styleProps.edge === 'end' && {
     marginRight: styleProps.size === 'small' ? -3 : -12,
   }),
+  '&:hover': {
+    '&.Mui-checked': {
+      'span:empty:not([class*=MuiTouchRipple-root])': {
+        backgroundColor: '#106ba3',
```

Comment:
What color is that? 

How does this style work in themes with a differen color scheme? Same applies to all the other color values.

Reply:
For all the colors I verbatim reused the colors from the individual files (ie CustomizedCheckbox.tsx) and moved them into the SwitchBase component instead. Happy though to update and utilize theme colors if you'd prefer it! (And that color is a slightly darker blue for the hover over a checked box or selected radio button)